rfcs/0000-execute-builds-in-standard-containers.md

```diff
@@ -0,0 +1,132 @@
+## Problem
+
+Currently, kpack executes each stage of the [CNB Lifecycle][cnb-lifecycle] inside of
+individual [init containers][init-containers]. This has two major benefits:
+
+1. Different stages of the lifecycle are isolated from one another allowing the credentials required to push to the
+   registry to be only mounted in the containers that need them, preventing a rogue [BuildPack][buildpack] from reading
+   the secrets.
+
+2. We can very easily define the order of the steps occurring as well as waiting for the previous step to complete
+   before continuing on. Additionally, failures in one init container cause the entire build to fail without additional
+   logic on our part.
+
+As kpack has grown, we have seen a need to support [service meshes][istio-service-mesh] such as [Istio][istio], which do
+not work when you have init containers that need to reach out to the network. This is
+a [well documented][istio-init-container-compatability] technical limitation related to
+Istio's [automatic sidecar injection][istio-sidecar-injection]. The limitation is due to Istio adding a sidecar
+container to the kpack build pod which runs [envoy proxy][envoy-proxy] and sets up traffic from the build containers to
+be routed through the proxy. Due to the fact that init containers run before any normal container can start, kpack
+builds are unable to reach the network because they are attempting to communicate through the envoy proxy container that
+has not and will not start until the build completes.
+
+To get around this issue, kpack currently applies an annotation (`sidecar.istio.io/inject=false`) to all build pods to
+disable Istio sidecar injection. This presents a problem where users might want to take advantage of the various
+features of Istio, but cannot because kpack is bypassing it.
+
+[buildpack]: https://buildpacks.io/
+
+[cnb-lifecycle]: https://buildpacks.io/docs/concepts/components/lifecycle/
+
+[envoy-proxy]: https://www.envoyproxy.io/
+
+[init-containers]: https://kubernetes.io/docs/concepts/workloads/pods/init-containers/
+
+[istio]: https://istio.io/
+
+[istio-sidecar-injection]: https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/#injection
+
+[istio-service-mesh]: https://istio.io/latest/about/service-mesh/#what-is-a-service-mesh
+
+[istio-init-container-compatability]: https://istio.io/latest/docs/setup/additional-setup/cni/#compatibility-with-application-init-containers
+
+## Outcome
+
+Kpack builds can execute successfully when an Istio sidecar when injected into a build pod.
+
+## Actions to take
```

Comment:
Step 3, calls out the feature flag but, is in conjunction with an implementation detail. Can we call that the user experience of the feature flag explicitly?